packages/aws-cdk-lib/aws-ec2/lib/instance.ts

```diff
@@ -702,7 +706,10 @@ export class Instance extends Resource implements IInstance {
       this.instance.cfnOptions.creationPolicy = {
         ...this.instance.cfnOptions.creationPolicy,
         resourceSignal: {
-          timeout: props.resourceSignalTimeout && props.resourceSignalTimeout.toIsoString(),
+          ...this.instance.cfnOptions.creationPolicy?.resourceSignal,
```

Comment:
Nice, just to confirm that this is needed otherwise it would remove `Count` from `ResourceSignal` right?

Reply:
There is surrounding if statement specified [here](https://github.com/aws/aws-cdk/blob/b6ad97f4b4e1c185ddc53f60e15b0dabd8022694/packages/aws-cdk-lib/aws-ec2/lib/instance.ts#L701)  that checks if `resourceSignalTimeout` is specified before updating the timeout